sdk/testing/src/main/java/io/opentelemetry/sdk/testing/assertj/OpenTelemetryAssertions.java

```diff
@@ -111,6 +117,107 @@ public static EventDataAssert assertThat(EventData eventData) {
+   * value satisfying {@code assertion}.
+   */
+  public static AttributeAssertion satisfies(
+      AttributeKey<Boolean> key, BooleanAssertConsumer assertion) {
+    return AttributeAssertion.create(key, assertion);
+  }
+
+  /**
+   * Returns an {@link AttributeAssertion} that asserts the given {@code key} is present with a
+   * value satisfying {@code assertion}.
+   */
+  public static AttributeAssertion satisfies(AttributeKey<Long> key, LongAssertConsumer assertion) {
+    return AttributeAssertion.create(key, assertion);
+  }
+
+  /**
+   * Returns an {@link AttributeAssertion} that asserts the given {@code key} is present with a
+   * value satisfying {@code assertion}.
+   */
+  public static AttributeAssertion satisfies(
+      AttributeKey<Double> key, DoubleAssertConsumer assertion) {
+    return AttributeAssertion.create(key, assertion);
+  }
+
+  /**
+   * Returns an {@link AttributeAssertion} that asserts the given {@code key} is present with a
+   * value satisfying {@code assertion}.
+   */
+  // Will require a cast only if using a custom implementation of AttributeKey which is highly
+  // unusual usage.
+  @SuppressWarnings("FunctionalInterfaceClash")
+  public static AttributeAssertion satisfies(
+      AttributeKey<List<String>> key, StringListAssertConsumer assertion) {
+    return AttributeAssertion.create(key, assertion);
+  }
+
+  /**
+   * Returns an {@link AttributeAssertion} that asserts the given {@code key} is present with a
+   * value satisfying {@code assertion}.
+   */
+  public static AttributeAssertion satisfies(
+      AttributeKey<List<Boolean>> key, BooleanListAssertConsumer assertion) {
+    return AttributeAssertion.create(key, assertion);
+  }
+
+  /**
+   * Returns an {@link AttributeAssertion} that asserts the given {@code key} is present with a
+   * value satisfying {@code assertion}.
+   */
+  public static AttributeAssertion satisfies(
+      AttributeKey<List<Long>> key, LongListAssertConsumer assertion) {
+    return AttributeAssertion.create(key, assertion);
+  }
+
+  /**
+   * Returns an {@link AttributeAssertion} that asserts the given {@code key} is present with a
+   * value satisfying {@code assertion}.
+   */
+  public static AttributeAssertion satisfies(
+      AttributeKey<List<Double>> key, DoubleListAssertConsumer assertion) {
+    return AttributeAssertion.create(key, assertion);
+  }
+
+  /**
+   * Returns an {@link AttributeAssertion} that asserts the given {@code key} is present with the
+   * given {@code value}.
+   */
+  public static <T> AttributeAssertion attribute(AttributeKey<T> key, T value) {
+    return AttributeAssertion.create(key, val -> val.isEqualTo(value));
+  }
+
+  // Unique interfaces to prevent generic functional interface clash. These are not interesting at
+  // all but are required to be able to use the same method name in methods like satisfies above.
+
+  public interface StringAssertConsumer extends Consumer<AbstractStringAssert<?>> {}
```

Comment:
Left these nested without javadoc since these are just types corresponding to lambdas and should never be directly referenced by users.